### ISSUE TITLE
SWTException: Invalid thread access in Memory view

### ISSUE THREAD

**Issue by jonahgraham** [eclipse-platform/eclipse.platform#710]
This is a regression in Eclipse 4.29 compared to 4.28 - when memory view is refreshed, common after editing a cell.

This is the stacktrace:

```java
!ENTRY org.eclipse.debug.ui 4 120 2023-09-20 16:20:57.444
!MESSAGE Error logged from Debug UI: 
!STACK 0
org.eclipse.swt.SWTException: Invalid thread access
	at org.eclipse.swt.SWT.error(SWT.java:4918)
	at org.eclipse.swt.SWT.error(SWT.java:4833)
	at org.eclipse.swt.SWT.error(SWT.java:4804)
	at org.eclipse.swt.widgets.Widget.error(Widget.java:565)
	at org.eclipse.swt.widgets.Widget.checkWidget(Widget.java:481)
	at org.eclipse.swt.widgets.Table.setRedraw(Table.java:3747)
	at org.eclipse.jface.viewers.StructuredViewer.refresh(StructuredViewer.java:1425)
	at org.eclipse.debug.internal.ui.memory.provisional.AbstractAsyncTableRendering.refresh(AbstractAsyncTableRendering.java:2107)
	at org.eclipse.debug.internal.ui.views.memory.renderings.AsyncTableRenderingUpdatePolicy.handleMemoryBlockChanged(AsyncTableRenderingUpdatePolicy.java:145)
	at org.eclipse.debug.internal.ui.views.memory.renderings.AsyncTableRenderingUpdatePolicy.modelChanged(AsyncTableRenderingUpdatePolicy.java:76)
	at org.eclipse.debug.internal.ui.viewers.provisional.AbstractModelProxy$1.run(AbstractModelProxy.java:92)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.debug.internal.ui.viewers.provisional.AbstractModelProxy.fireModelChanged(AbstractModelProxy.java:96)
	at org.eclipse.debug.internal.ui.viewers.update.MemoryBlockProxy$1.handleChange(MemoryBlockProxy.java:54)
	at org.eclipse.debug.internal.ui.viewers.update.EventHandlerModelProxy.dispatchChange(EventHandlerModelProxy.java:237)
	at org.eclipse.debug.internal.ui.viewers.update.EventHandlerModelProxy.handleDebugEvents(EventHandlerModelProxy.java:137)
	at org.eclipse.debug.core.DebugPlugin$EventNotifier.run(DebugPlugin.java:1225)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.debug.core.DebugPlugin$EventNotifier.dispatch(DebugPlugin.java:1259)
	at org.eclipse.debug.core.DebugPlugin$EventDispatchJob.run(DebugPlugin.java:491)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

This can easily be replicated with CDT by opening memory view, adding a memory monitor, opening a hex rendering and editing a cell. 

This can also be replicated using the platform debug example PDA launch type in a dev environment

1. Add the debug example bundles to your workspace (all of the bundles with examples in their name in the [repo](https://github.com/eclipse-platform/eclipse.platform/tree/master/debug))
2. Launch a runtime workbench
3. Create a simple project which has the [PDA samples](https://github.com/eclipse-platform/eclipse.platform/tree/master/debug/org.eclipse.debug.examples.core/samples)
4. Create a `PDA Application` debug launch configuration for `counter.pda`
5. Launch in debug mode
6. Press pause
7. Open Memory View
8. Press Add (next to Monitors label)
9. Fill in 100 for address and 100 for number of bytes
10. (maybe needed, seems like a different bug) change to a different view in same tab group and back again to display the rendering
11. click in first cell and type a new number
12. press Enter
13. Observe error log has errors like above

Here is a screen recording as I am sure that some people may not know about this:

https://github.com/eclipse-platform/eclipse.platform.ui/assets/679236/ea0d8bcb-3b0b-433d-9343-ec3d0a6da6e6

**Comment by Kummallinen** [comment]
A comment on the severity of this bug, it makes 2023-09 essentially unstable for anyone who's workflow uses the memory view. Performing any edit or a memory import currently requires a manual refresh before the edited memory is shown. It is also silent unless you have the error log open so users may not realise it is a rendering issue.

I am aware there is no process for building a patch release for users. But could the fix be backported to the maintenance branch? This would make it easier for those to build their own products on top of Eclipse to pick up a fix. Especially as there are other important fixes in SimRel 2023-09 & Platform 4.29 that make updating more important than usual.

**Comment by laeubi** [comment]
> But could the fix be backported to the maintenance branch?

Everyone can create PRs targeting any branch.

> This would make it easier for those to build their own products on top of Eclipse to pick up a fix

Of course one can also simply reference the I-Build repository until it is fully released.
